### DISCUSSION TITLE
Detail of graph varational autoencoder KL loss

### DISCUSSION THREAD

**Discussion by xiachenrui** [pyg-team/pytorch_geometric#6296] [Q&A]
Hi, all

I found the PyG give an example of GVAE, and it calculated KL loss in the following  [way ](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/autoencoder.py)
```
    if args.variational:
        loss = loss + (1 / train_data.num_nodes) * model.kl_loss()
```
But I think it show be divided  by the dimension of node features rather than the node numbers , such as

```
    if args.variational:
        loss = loss + (1 / train_data.data.shape[1]) * model.kl_loss()
```

Please let me know your ideas

Best !

Chenrui

**Comment by rusty1s** [comment]
This is directly taken from [here](https://github.com/tkipf/gae/blob/master/gae/optimizer.py#L33). Since the KL loss is operating on a dense N^2 matrix, I think this is correct.
